sympy/core/numbers.py

```diff
@@ -417,8 +418,11 @@ def __rdivmod__(self, other):
             raise TypeError(msg % (type(other).__name__, type(self).__name__))
         return divmod(other, self)
 
-    def __round__(self, *args):
-        return round(float(self), *args)
+    def __round__(self,ndigits = 0):
+        self = str(self)
+        index = self.index(".")
+        self = self[:index+1+ndigits]
+        return Decimal(self)
```

Comment:
Please use the `Float` algorithms, not the decimal ones. `Float.evalf` is (almost) what you need here.